### DISCUSSION TITLE
How to get the history of issue, like the history of due date or assignee

### DISCUSSION THREAD

**Discussion by guoguoguilai** [python-gitlab/python-gitlab#1507] [Q&A] [answered]
## Description of the problem, including code/CLI snippet
How to get the history of issue, like the history of due date or assignee

## Expected Behavior
Get the history of issues.

## Actual Behavior
Cannot find the corresponding API

## Specifications

  - python-gitlab version:  2.7.1
  - API version you are using (v3/v4): v4 
  - Gitlab server version (or gitlab.com): 13.8.3-ee

**Comment by nejch** [comment]
I don't think this is supported by the API, see https://docs.gitlab.com/ee/api/issues.html#single-issue, so python-gitlab will not be able to fetch this. You might have more luck by exporting the project and inspecting the `ndjson` files in case this is stored in the archives.

**Comment by guoguoguilai** [comment]
> I don't think this is supported by the API, see https://docs.gitlab.com/ee/api/issues.html#single-issue, so python-gitlab will not be able to fetch this. You might have more luck by exporting the project and inspecting the `ndjson` files in case this is stored in the archives.

Thanks for your quickly replying. Can you show more details on "exporting the project and inspecting the ndjson files"?

**Comment by nejch** [comment, marked as answer]
You can export the entire project using https://python-gitlab.readthedocs.io/en/stable/gl_objects/projects.html#import-export.

This will give you a tarball with the following structure: https://docs.gitlab.com/ee/development/import_export.html#project

Look for `./tree/project/issues.ndjson` and see if the historical data is there. The ndjson is not very readable but give it a try. But I can't promise anything, this is really up to GitLab whether they store the information, I think it's very limited, but the EE version might actually have everything.

**Reply by nejch** [reply]
I've turned this into a Q&A discussion as I dont think we can help with python-gitlab alone, but I'd be interested if you found a solution, let us know @guoguoguilai :)
